src/librustdoc/passes/collect_intra_doc_links/early.rs

```diff
@@ -32,11 +32,6 @@ crate fn early_resolve_intra_doc_links(
         all_trait_impls: Default::default(),
     };
 
-    // Because of the `crate::` prefix, any doc comment can reference
-    // the crate root's set of in-scope traits. This line makes sure
-    // it's available.
-    loader.add_traits_in_scope(CRATE_DEF_ID.to_def_id());
```

Comment:
Can you explain to me the rationale of why this is no longer necessary (i.e., what the new behavior is) with the new way `crate::` is resolved? I partially understand, but I want to make sure I understand fully.

Reply:
`crate::Type::AssocItem` now takes the set of traits in scope at the point where it's written, and not elsewhere (in the crate root module). This is the change that fixes https://github.com/rust-lang/rust/issues/78696.

So we no longer need to collect traits in scope at arbitrary root modules.

(The old buggy behavior was caused by replacing `module_id` with `CRATE_DEF_ID` too early.)

Reply:
> (The old buggy behavior was caused by replacing module_id with CRATE_DEF_ID too early.)

Ah, thanks, I understand now.